### PR TITLE
feature: add spanner support

### LINKED ISSUE #210: Support GCP Spanner with GoogleSQL dialect
I'm a big fan of `testfixtures`. It allowed me to implement integration tests with PostgreSQL and MySQL. Now that our team has adopted Spanner, we are missing the convenience `testfixtures` provide to generate test data declaratively.

We can use github.com/googleapis/go-sql-spanner which is compatible with `database/sql`.

### PR DESCRIPTION
**Pull request by kikihakiem**
Added support for GCP Spanner with GoogleSQL dialect. Closes #210

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,6 +4,7 @@ RUN apk update
 RUN apk add alpine-sdk
 
 WORKDIR /testfixtures
-COPY . .
+COPY go.* .
 
 RUN go mod download
+COPY . .
```
Comment (kikihakiem) [inline Dockerfile]: speed up `task docker`

File: docker-compose.yml [hunk 2]
```diff
@@ -50,3 +53,9 @@ services:
       CLICKHOUSE_DB: testfixtures_test
       CLICKHOUSE_USER: clickhouse
       CLICKHOUSE_PASSWORD: clickhouse
+
+  spanner:
+    image: gcr.io/cloud-spanner-emulator/emulator
+    ports:
+      - "19010:9010"
```
Comment (kikihakiem) [inline docker-compose.yml:60]: `9010` already bound by Clickhouse

File: spanner_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	"testing"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"cloud.google.com/go/spanner/admin/instance/apiv1/instancepb"
+)
+
+func TestSpanner(t *testing.T) {
+	prepareSpannerDB(t)
```
Comment (kikihakiem) [inline spanner_test.go:19]: necessary for creating Spanner instance and the DB

File: spanner.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+func (*spanner) paramType() int {
+	return paramTypeAtSign
+}
+
+func (*spanner) quoteKeyword(str string) string {
+	return str
+}
+
+func (*spanner) databaseName(q queryable) (string, error) {
+	return "testdb", nil
```
Comment (kikihakiem) [inline spanner.go]: there's no easy way to get the database name. How about making the `skipTestDatabaseCheck` mandatory when using "spanner" dialect by returning an error here? The message could be "we cannot check whether you're using a test database. Please ensure that you're using a test database and set `skipTestDatabaseCheck` to true" or something similar.

File: Dockerfile (hunk 1 above)
Comment (slsyy) [reply Dockerfile]: i did something similiar here https://github.com/go-testfixtures/testfixtures/pull/204

File: spanner.go (hunk 4 above)
Comment (slsyy) [reply spanner.go]: could you return an error instead, which indicate that `skipTestDatabaseCheck` is impossible on Spanner, so users need to include `testfixtures.DangerousSkipTestDatabaseCheck()` purposely? 

Also it should be documented in a `README.md`

File: spanner.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	referencingTable string
+	foreignKeyColumn string
+	referenceTable   string
+	referenceColumn  string
+}
+
+func (h *spanner) init(db *sql.DB) error {
+	var err error
+
+	if h.cleanTableFn == nil {
```
Comment (slsyy) [inline spanner.go:28]: `init` is called once in a `New`, you don't need this check

File: spanner.go [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+type spannerConstraint struct {
+	constraintName   string
+	referencingTable string
+	foreignKeyColumn string
+	referenceTable   string
+	referenceColumn  string
+}
+
+func (h *spanner) init(db *sql.DB) error {
+	var err error
```
Comment (slsyy) [inline spanner.go]: could you move it line above where it is needed?

File: spanner_test.go [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+		os.Getenv("SPANNER_CONN_STRING"),
+		"testdata/schema/spanner.sql",
+	)
+}
+
+func prepareSpannerDB(t *testing.T) {
+	t.Helper()
+
+	var err error
+	if err = os.Setenv("SPANNER_EMULATOR_HOST", "spanner:9010"); err != nil {
```
Comment (slsyy) [inline spanner_test.go]:
```suggestion
	if err := os.Setenv("SPANNER_EMULATOR_HOST", "spanner:9010"); err != nil {
```

**Comment by slsyy** [comment]
Thank you @kikihakiem

**Comment by slsyy** [comment]
@kikihakiem do you want a new release?

**Comment by kikihakiem** [comment]
hi @slsyy . Currently, we specify the commit SHA in our go.mod files, but I don't mind a new release

**Comment by slsyy** [comment]
@kikihakiem released in https://github.com/go-testfixtures/testfixtures/releases/tag/v3.13.0

**Pull request merged**